File: src/main/java/com/baulsupp/oksocial/services/surveymonkey/SurveyMonkeyServiceDefinition.java
package com.baulsupp.oksocial.services.surveymonkey;

import com.baulsupp.oksocial.credentials.ServiceDefinition;
import java.util.Optional;

import static java.util.Optional.of;

public class SurveyMonkeyServiceDefinition implements ServiceDefinition<SurveyMonkeyToken> {
  @Override public String apiHost() {
    return "api.surveymonkey.net";
  }

  @Override public String serviceName() {
    return "Survey Monkey API";
  }

  @Override public String shortName() {
    return "surveymonkey";
  }

  public SurveyMonkeyToken parseCredentialsString(String s) {
    String[] parts = s.split(":", 2);
    return new SurveyMonkeyToken(parts[0], parts[1]);
  }

  public String formatCredentialsString(SurveyMonkeyToken credentials) {
    return credentials.apiKey + ":" + credentials.accessToken;
  }

  @Override public Optional<String> apiDocs() {
    return of("https://developer.surveymonkey.com/api/v3/#scopes");
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/surveymonkey/SurveyMonkeyServiceDefinition.kt
package com.baulsupp.oksocial.services.surveymonkey

import com.baulsupp.oksocial.credentials.ServiceDefinition

class SurveyMonkeyServiceDefinition : ServiceDefinition<SurveyMonkeyToken> {
  override fun apiHost(): String = "api.surveymonkey.net"

  override fun serviceName(): String = "Survey Monkey API"

  override fun shortName(): String = "surveymonkey"

  override fun parseCredentialsString(s: String): SurveyMonkeyToken {
    val parts = s.split(":".toRegex(), 2).toTypedArray()
    return SurveyMonkeyToken(parts[0], parts[1])
  }

  override fun formatCredentialsString(credentials: SurveyMonkeyToken) =
      credentials.apiKey + ":" + credentials.accessToken

  override fun apiDocs() = "https://developer.surveymonkey.com/api/v3/#scopes"

  override fun accountsLink() = "https://developer.surveymonkey.com/apps/"
}
